From 87e024825b50ad76a49524a08a6be6368a66172a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?G=C3=BCnther=20Deschner?= <gd@samba.org>
Date: Fri, 29 Apr 2016 13:28:42 +0200
Subject: [PATCH 1079/1096] s3:client:smbspool_krb5_wrapper: fix the non
 clearenv build.

Guenther

Signed-off-by: Guenther Deschner <gd@samba.org>
Reviewed-by: Volker Lendecke <vl@samba.org>

Autobuild-User(master): Volker Lendecke <vl@samba.org>
Autobuild-Date(master): Fri Apr 29 19:52:23 CEST 2016 on sn-devel-144

(cherry picked from commit e0d8c6b6b428819206fb6e96c47e952845af0874)

Bug: https://bugzilla.samba.org/show_bug.cgi?id=11864
(cherry picked from commit 290541a43a94d806ff13866439b0326763b468d4)
---
 source3/client/smbspool_krb5_wrapper.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/source3/client/smbspool_krb5_wrapper.c b/source3/client/smbspool_krb5_wrapper.c
index 89db5e0..d26a7a6 100644
--- a/source3/client/smbspool_krb5_wrapper.c
+++ b/source3/client/smbspool_krb5_wrapper.c
@@ -31,6 +31,8 @@
 
 #include "dynconfig/dynconfig.h"
 
+#undef calloc
+
 enum cups_smb_dbglvl_e {
 	CUPS_SMB_LOG_DEBUG = 0,
 	CUPS_SMB_LOG_ERROR,
@@ -199,8 +201,10 @@ int main(int argc, char *argv[])
 #ifdef HAVE_CLEARENV
 	clearenv();
 #else
-	extern char **environ;
-	environ = calloc(1, sizeof(*environ));
+	{
+		extern char **environ;
+		environ = calloc(1, sizeof(*environ));
+	}
 #endif
 
 	CUPS_SMB_DEBUG("Setting KRB5CCNAME to '%s'", gen_cc);
-- 
2.9.0

